streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableGlobalKTableLeftJoinTest.java

```diff
@@ -0,0 +1,173 @@
+        assertThat(a.oldValue, is(nullValue()));
+    }
+
+    @Test
+    public void shouldSendNewAndOldValuesCorrectly() throws Exception {
+        join.enableSendingOldValues();
+        final Processor<String, Change<String>> processor = join.get();
+        processor.init(context);
+        global.put("1", "B");
+        global.put("2", "C");
+        processor.process("A", new Change<>("2", "1"));
+        final Change<String> a = (Change<String>) context.forwardedValues.get("A");
+        assertThat(a.newValue, equalTo("2+C"));
+        assertThat(a.oldValue, equalTo("1+B"));
+    }
+
+    @Test
+    public void shouldSendOldValueIfNewValueIsNullOldValueIsntNull() throws Exception {
+        join.enableSendingOldValues();
+        final Processor<String, Change<String>> processor = join.get();
+        processor.init(context);
+        global.put("1", "B");
+        processor.process("A", new Change<>(null, "1"));
+        final Change<String> a = (Change<String>) context.forwardedValues.get("A");
+        assertThat(a.newValue, is(nullValue()));
+        assertThat(a.oldValue, equalTo("1+B"));
+    }
+
+    @Test
+    public void shouldJoinAndForwardIfOldKeyNotInOtherStoreAndSendOldValues() throws Exception {
+        join.enableSendingOldValues();
+        final Processor<String, Change<String>> processor = join.get();
+        processor.init(context);
+        processor.process("A", new Change<>(null, "1"));
+        final Change<String> a = (Change<String>) context.forwardedValues.get("A");
+        assertThat(a.newValue, is(nullValue()));
+        assertThat(a.oldValue, is("1+null"));
+    }
+
+    @Test
+    public void shouldNotForwardIfOtherStoreValueIsNullAndNewAndOldValuesAreNull() throws Exception {
+        join.enableSendingOldValues();
+        final Processor<String, Change<String>> processor = join.get();
+        processor.init(context);
+        processor.process("A", new Change<String>(null, null));
+        assertThat(context.forwardedValues.get("A"), is(nullValue()));
+    }
+
+    @Test
+    public void shouldSendDeletesIfChangeHasOldValue() throws Exception {
+        global.put("1", "A");
+        final Processor<String, Change<String>> processor = join.get();
+        processor.init(context);
+        processor.process("A", new Change<>(null, "1"));
+        final Change<String> a = (Change<String>) context.forwardedValues.get("A");
+        assertThat(a.newValue, is(nullValue()));
+    }
+
+    @Test
+    public void shouldNotSendAnythingIfChangeIsNullNullAndKeyMapsToNullInOtherTable() throws Exception {
+        global.put("1", "A");
+        final Processor<String, Change<String>> processor = join.get();
+        processor.init(context);
+        processor.process("A", new Change<String>(null, null));
+        assertThat(context.forwardedValues.get("A"), is(nullValue()));
+    }
+
+    @Test
+    public void shouldNotForwardIfBothNewAndOldValuesAreNull() throws Exception {
```

Comment:
What is the different from this test to `shouldNotSendAnythingIfChangeIsNullNullAndKeyMapsToNullInOtherTable`?

Furhtermore, why do you setup a new `KTableGlobalKTableLeftJoin` ? If I did not miss anything, it is the same setup as the global member `join` variable.